### DISCUSSION TITLE
Using your own tag model

### DISCUSSION THREAD

**Discussion by nargalzius** [spatie/laravel-tags#327] [Q&A]
So I followed the section on using your own tag model to properly use a specific connection on my multi-db connection.

However, we're using uuids in stead of ids for our primary keys, and I can't seem to find documentation on how to adjust it to recognize and use taggable_uuid.

Some guidance would be much appreciated.

**Comment by AliSafaei1990** [comment]
if the model that you are attaching the tag to it has uuid, so you need to change database/migration/2025_00_00_000000_create_tag_tables .php : 
`        Schema::create('taggables', function (Blueprint $table) {
            $table->foreignId('tag_id')->constrained()->cascadeOnDelete();


            $table->uuidMorphs('taggable'); // change this line

            $table->unique(['tag_id', 'taggable_id', 'taggable_type']);
        });`

**Comment by nargalzius** [comment]
This is the actual up function.

When you say change the line... you mean the $able->uuid('taggable_id')?
Or do I add it?

What should the up codeblock look like with the working changes?

```
public function up()
    {
        Schema::create('tags', function (Blueprint $table) {
            $table->id();
            $table->json('name');
            $table->json('slug');
            $table->string('type')->nullable();
            $table->integer('order_column')->nullable();
            $table->timestamps();
        });

        Schema::create('taggables', function (Blueprint $table) {
            $table->foreignId('tag_id')->constrained()->cascadeOnDelete();
            $table->string('taggable_type');
            $table->uuid('taggable_id');
            $table->unique(['tag_id', 'taggable_id', 'taggable_type']);
            
        });
    }
```

Or am I correct to assume these two lines I have
```
$table->string('taggable_type');
$table->uuid('taggable_id');
```
should be changed the single one below?
```
$table->uuidMorphs('taggable');
```

**Reply by jonjakoblich** [reply]
Yes, you can do it that way.
